### PR TITLE
Use nmake

### PR DESCRIPTION
**Pull request by nizar-sallem**
@dirk-thomas hope you are doing well.
This is just an enhancement that would allow handling nmake same way as ninja.
Did only test with indigo but can compare with latest development and update my PR

### CODE REVIEW

File: python/catkin/builder.py [hunk 1]
```diff
@@ -403,10 +403,13 @@ def build_catkin_package(
     else:
         print('%s exists, skipping explicit cmake invocation...' % makefile_name)
         # Check to see if cmake needs to be run via make
-        if not use_ninja:
-            make_check_cmake_cmd = ['make', 'cmake_check_build_system']
-        else:
+        if use_ninja:
             make_check_cmake_cmd = ['ninja', 'build.ninja']
+        elif use_namke:
```
Comment (dirk-thomas) [inline python/catkin/builder.py]: Spelling error fails CI.

**Comment by dirk-thomas** [comment]
Thanks for the pull request - good to see you continue to work with ROS.

Once this passes CI I would merge it into kinetic-devel first. It can also be backported into Indigo / Jade if you need that since the risk of regressions seems rather low to me.

File: python/catkin/builder.py (hunk 1 above)
Comment (nizar-sallem) [inline python/catkin/builder.py]: Fixing

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder retest this please

**Comment by dirk-thomas** [comment]
I squashed and cherry-picked the patch to the _kinetic-devel_ branch: d0c4ecb3bac9b38a0883253e1e350d3fb72c4a6f

**Comment by nizar-sallem** [comment]
Awesome, thanks

File: python/catkin/builder.py [hunk 2]
```diff
@@ -520,9 +530,11 @@ def build_cmake_package(
 
     # Run make
     if not use_ninja:
-        make_executable = 'make'
-    else:
         make_executable = 'ninja'
```
Comment (gavanderhoorn) [inline python/catkin/builder.py:533]: Is this condition inverted? It looks like it now says:

``` python
if not use_ninja:
    make_executable = 'ninja'
```

That would seem to be the inverse of what is actually the intent.

Comment (dirk-thomas) [reply python/catkin/builder.py:533]: For reference: #826

**Pull request closed without merging**